clang/include/clang/Basic/Attributes.h

```diff
@@ -23,6 +23,30 @@ int hasAttribute(AttributeCommonInfo::Syntax Syntax,
                  const IdentifierInfo *Scope, const IdentifierInfo *Attr,
                  const TargetInfo &Target, const LangOptions &LangOpts);
 
+inline const char* deuglifyAttrScope(StringRef Scope) {
```

Comment:
not a huge fan of `ugly` here as it isn't particularly descriptive, but mostly just on the name of it.